scylla/src/transport/topology.rs

```diff
@@ -177,6 +177,10 @@ async fn query_peers(conn: &Connection, connect_port: u16) -> Result<Vec<Peer>,
     let typed_local_rows =
         local_rows.into_typed::<(IpAddr, Option<String>, Option<String>, Option<Vec<String>>)>();
 
+    // Here we are considering the local node to be just a regular peer, but datastax claims that unless
+    // you have SNI enabled, you shouldn't trust rpc_address (So they just use the initial connection's address)
+    // we could expose this from conn & use it directly instead of system.local.rpc_address
+
```

Comment:
While usage of `listen_address` was clearly wrong and `rpc_address` is better, I don't understand the issue related to disabled SNI. Could you explain it, or at least point to some resources with an explanation?

Reply:
https://github.com/datastax/nodejs-driver/blob/master/lib/control-connection.js#L716

I'm just referencing what is done in data stack's driver. They are ignoring the result entirely unless the client specifies sni here. (And using the initial connection's address)